src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -338,10 +341,24 @@
                                         }
                                     }
                                     Err(err) => {
-                                        warn!(
-                                            "OpenID Connect: Could not extend login session for user '{}': {}",
-                                            &session.id, err
-                                        );
+                                        match &err {
+                                            // this is where the RFC-6749 5.2 Error Response is received and
+                                            // return to the caller. It's the responsibility of the caller
+                                            // to decide whether to retry or report back to the user.
+                                            openidconnect::RequestTokenError::ServerResponse(r) => {
+                                                warn!(
+                                                    "OpenID Connect: Could not extend login session for user '{}': {}",
+                                                    &session.id, "Connection to OpenID Connect provider not yet established"
+                                                );
+                                                return Err(r.error().clone());
+                                            }
+                                            _ => {
+                                                return Err(CoreErrorResponseType::Extension(
```

Comment:
None of the information about the actual error is returned here, is there nothing of value to pass back about the error?

Reply:
See my comment above, I need to figure out more details about the other variants here